### PR TITLE
 new param: new_ticket_threshold

### PR DESCRIPTION
**Pull request by Bonko**
if an open ticket with a different service description for the host is found only post to it if last edit is time is < new_ticket_threshold minutes.
otherwise create a new ticket

### CODE REVIEW

File: nagios2trac.py [hunk 1]
```diff
@@ -52,6 +50,12 @@ def debug_output(output):
 if options.description is None and not options.listmethods:
     parser.error("please specify a scription")
 
+# prefer cli option over configfile
+if options.new_ticket_threshold is not None:
+    new_ticket_threshold=options.new_ticket_threshold
+else:
+    new_ticket_threshold=trac_new_ticket_threshold
```
Comment (mlafeldt) [inline nagios2trac.py]: Simpler:

```
new_ticket_threshold = options.new_ticket_threshold or trac_new_ticket_threshold
```

File: nagios2trac.py [hunk 2]
```diff
@@ -118,24 +132,29 @@ def debug_output(output):
...
+        current_time_utc=datetime.datetime.utcnow()
 
+        current_time_minus_threshold=current_time_utc-datetime.timedelta(minutes=new_ticket_threshold)
+
+        if last_modified_utc < current_time_minus_threshold:
+            debug_output("creating a new ticket because old one has not been modified for more than configured threshold value (%d) minutes" % new_ticket_threshold)
+            create_ticket()
+        else:
+            # maybe only post if last edit time > 15 min to prevent trac spam when many services of a host fail
+            server.ticket.update(open_ticket_for_same_host[0], comment_template,{},trac_notifications)
```
Comment (mlafeldt) [inline nagios2trac.py]: Maybe add a update_ticket function?

**Comment by mlafeldt** [comment]
Looks better now.

**Comment by nightlyone** [comment]
@Bonko Would be nice, if you can attach the unit of the time to the variable (name) or the value.
Like new_ticket_threshold_seconds

**Comment by Bonko** [comment]
the value is stated in the help and in the default configuration, but good idea to clarify

**Pull request merged**